### ISSUE TITLE
Rename nv -> nvertices and ne -> nedges

### ISSUE THREAD

**Issue by AzamatB** [JuliaGraphs/LightGraphs.jl#1118]
Would be more descriptive, and hence more Julian, IMO.

**Comment by AzamatB** [comment]
Also for consideration:
`all_neighbors` -> `allneighbors` (would be consistent with existing `inneighbors `, `outneighbors`)
`is_directed` -> `isdirected` (all methods in Base starting with `is` are written together)

**Comment by sbromberger** [comment]
1) `nv` and `ne` are pretty standard and are frequently used, so should be as simple as possible.
2) see https://github.com/JuliaGraphs/LightGraphs.jl/issues/828 for previous discussions on `*_neighbors` and `*_directed`.

**Comment by sbromberger** [comment]
See also https://discourse.julialang.org/t/naming-remove-all-underscores-to-matter-what/8549/59.

**Comment by AzamatB** [comment]
> If it is two words and the first one is not {in,out}, then use an underscore.

This convention seems rather arbitrary and hard to justify... IMO, In such situations it is best to follow [Julia's style guide](https://docs.julialang.org/en/v1/manual/style-guide/#Use-naming-conventions-consistent-with-Julia-base/-1), which is also arbitrary. But at least this way user has to remember only 1 random convention, instead of current situation where he has another random convention in addition to Base's creating additional cognitive burden that otherwise can be avoided.
The fact that we write `inneighbors`, but `all_neighbors` just seems bad to me...

**Comment by sbromberger** [comment]
Our convention predates the style guide, I think, even if the formalization does not. 

In any case, this would have to wait until our next major version, with a period of deprecation in between. We could consider it once we are closer to that point. I am not opposed.

**Comment by juliohm** [comment]
I agree with @AzamatB :+1: it would be nice to have the underscores gone.
